crates/astria-auctioneer/src/metrics.rs

```diff
@@ -1,25 +1,220 @@
-use telemetry::{
+use astria_telemetry::{
     metric_names,
     metrics::{
         self,
+        Counter,
+        Histogram,
+        IntoF64,
         RegisteringBuilder,
     },
 };
 
-pub struct Metrics {}
+const AUCTION_BIDS_LABEL: &str = "auction_bids";
+const AUCTION_BIDS_PROCESSED: &str = "processed";
+const AUCTION_BIDS_DROPPED: &str = "dropped";
 
-impl Metrics {}
+const AUCTION_WINNER_SUBMISSION_LATENCY_LABEL: &str = "auction_winner_submission_latency";
+const AUCTION_WINNER_ERROR: &str = "error";
+const AUCTION_WINNER_SUCCESS: &str = "success";
 
-impl metrics::Metrics for Metrics {
+pub struct Metrics {
+    auction_bid_delay_since_start: Histogram,
+    auction_bids_dropped_histogram: Histogram,
+    auction_bids_processed_histogram: Histogram,
+    auction_bids_received_count: Counter,
+    auction_bids_without_matching_auction: Counter,
+    auction_winner_submission_error_latency: Histogram,
+    auction_winner_submission_success_latency: Histogram,
+    auction_winning_bid_histogram: Histogram,
+    auctions_cancelled_count: Counter,
+    auctions_submitted_count: Counter,
+    block_commitments_received_count: Counter,
+    executed_blocks_received_count: Counter,
+    optimistic_blocks_received_count: Counter,
+}
+
+impl Metrics {
+    pub(crate) fn increment_auction_bids_received_counter(&self) {
+        self.auction_bids_received_count.increment(1);
+    }
+
+    pub(crate) fn increment_auctions_cancelled_count(&self) {
+        self.auctions_cancelled_count.increment(1);
+    }
+
+    pub(crate) fn increment_auctions_submitted_count(&self) {
+        self.auctions_submitted_count.increment(1);
+    }
+
+    pub(crate) fn increment_auction_bids_without_matching_auction(&self) {
+        self.auction_bids_without_matching_auction.increment(1);
+    }
+
+    pub(crate) fn increment_block_commitments_received_counter(&self) {
+        self.block_commitments_received_count.increment(1);
+    }
+
+    pub(crate) fn increment_executed_blocks_received_counter(&self) {
+        self.executed_blocks_received_count.increment(1);
+    }
+
+    pub(crate) fn increment_optimistic_blocks_received_counter(&self) {
+        self.optimistic_blocks_received_count.increment(1);
+    }
+
+    pub(crate) fn record_auction_bids_processed_histogram(&self, val: impl IntoF64) {
+        self.auction_bids_processed_histogram.record(val);
+    }
+
+    pub(crate) fn record_auction_bids_dropped_histogram(&self, val: impl IntoF64) {
+        self.auction_bids_dropped_histogram.record(val);
+    }
+
+    pub(crate) fn record_auction_bid_delay_since_start(&self, val: impl IntoF64) {
+        self.auction_bid_delay_since_start.record(val);
+    }
+
+    pub(crate) fn record_auction_winning_bid_histogram(&self, val: impl IntoF64) {
+        self.auction_winning_bid_histogram.record(val);
+    }
+
+    pub(crate) fn record_auction_winner_submission_error_latency(&self, val: impl IntoF64) {
```

Comment:
we also might want the number of times submission has error'ed out. We could use this for an alert to indicate when auctioneer has a high submission error rate which is something actionable?

Reply:
Isn't that covered by the same histogram with the "error" label (see the method just below this one)?

Reply:
yes, the histogram metric will return a total count too. so that should cover it!